sources/mongodb/helpers.py

```diff
@@ -0,0 +1,122 @@
+"""SQL database source helpers"""
+
+import json
+from datetime import datetime
+from itertools import islice
+from typing import Any, Dict, Iterator, Optional
+
+import dlt
+from bson.decimal128 import Decimal128
+from bson.objectid import ObjectId
+from dlt.common.configuration.specs import BaseConfiguration, configspec
+from dlt.common.typing import TDataItem
+from pymongo import MongoClient
+from pymongo.collection import Collection
+from pymongo.cursor import Cursor
+
+CHUNK_SIZE = 10000
+
+
+class CollectionLoader:
+    def __init__(
+        self,
+        client: MongoClient[Any],
+        collection: Collection[Any],
+        incremental: Optional[dlt.sources.incremental[Any]] = None,
+    ) -> None:
+        self.client = client
+        self.collection = collection
+        self.incremental = incremental
+        if incremental:
+            self.cursor_field = incremental.cursor_path
+            self.last_value = incremental.last_value
+        else:
+            self.cursor_column = None
+            self.last_value = None
+
+    def make_query(self) -> Cursor[Any]:
+        if not self.incremental or not self.last_value:
+            filter_op = {}
+        elif (
+            self.incremental.last_value_func is max
+        ):  # Filtered according to last_value function
+            filter_op = {self.cursor_field: {"$gte": self.last_value}}
+        elif self.incremental.last_value_func is min:
+            filter_op = {self.cursor_field: {"$lt": self.last_value}}
+        else:  # Custom last_value, load everything and let incremental handle filtering
+            filter_op = {}
+        cursor = self.collection.find(filter_op)
+        return cursor
+
+    def clean_document(self, document: Dict[str, Any]) -> Any:
+        """Clean document to be json serializable"""
+        output = {}
+        if not isinstance(document, dict):
+            return document
+        for key, value in document.items():
+            if isinstance(value, dict):
+                output[key] = self.clean_document(value)
+            elif isinstance(value, list):
+                output[key] = [self.clean_document(item) for item in value]
+            elif isinstance(value, Decimal128):
```

Comment:
I hope there are no more decimals like decimal256 etc. used by mongo? maybe there's just generic type decimal in bson?

Reply:
I checked it, and `Decimal128` is the only decimal in [bson](https://github.com/mongodb/mongo-python-driver/blob/master/bson/__init__.py#L136).